src/main/java/de/progen_bot/listeners/SayListener.java

```diff
@@ -0,0 +1,89 @@
+package de.progen_bot.listeners;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.awt.Color;
+
+import net.dv8tion.jda.api.events.message.priv.PrivateMessageReceivedEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+
+public class SayListener extends ListenerAdapter {
+    private static Map<String, SayModel> sayStorage = new HashMap<>();
+
+    @Override
+    public void onPrivateMessageReceived(PrivateMessageReceivedEvent event) {
+        String userId = event.getAuthor().getId();
+
+        if (sayStorage.containsKey(userId)) {
+
+            if (event.getMessage().getContentRaw().equals("abort")) {
+                sayStorage.remove(userId);
+                return;
+            }
+
+            SayModel mSayModel = sayStorage.get(userId);
+            switch (mSayModel.getStage()) {
+                case 0:
+                    if (isHexadecimal(event.getMessage().getContentRaw())) {
+
+                        mSayModel.setEmbedBuilder(
+                                mSayModel.getEmbedBuilder().setColor(Color.decode(event.getMessage().getContentRaw())));
+                        mSayModel.setStage(1);
+                        // Replace old sayModel in every stage
+                        sayStorage.replace(userId, mSayModel);
+                        event.getChannel().sendMessage("Please enter a title for your embed.").queue();
+                    } else {
+                        event.getChannel().sendMessage("This is not a valid color!").queue();
+                    }
+                    break;
+                case 1:
+                    mSayModel.setEmbedBuilder(mSayModel.getEmbedBuilder().setTitle(event.getMessage().getContentRaw()));
+                    mSayModel.setStage(2);
+                    sayStorage.replace(userId, mSayModel);
+                    event.getChannel().sendMessage("Do you want to set an url for the title? <y/n>").queue();
+                    break;
+                case 2:
+                    if (event.getMessage().getContentRaw().equals("y")) {
+                        sayStorage.replace(userId, mSayModel);
+                        mSayModel.setStage(3);
```

Comment:
Shouldn't it be reversed? So updating the model will consult in updating the map?